curriculum/challenges/english/20-upcoming-python/learn-string-manipulation-by-building-a-cipher/65521203d7165c7b84b22ad4.md

```diff
@@ -0,0 +1,43 @@
+---
+id: 65521203d7165c7b84b22ad4
+title: Step 18
+challengeType: 20
+dashedName: step-18
+---
+
+# --description--
+
+Repeating all this process for each character in `text` would be really annoying. Fortunately, you can use a loop for that. A loop enables you to iterate through a sequence of elements and perform actions on each element.
+Now you are going to use a `for` loop. The general syntax is:
+
+```py
+for i in iterable:
+    <code>
+```
+
+`i` sequentially takes the value of the elements in `iterable`. It can have any valid name, but it's convenient to give it a meaningful name. `iterable` is the sequence you want to iterate over. Last but not least, the code to execute at each iteration consitutes the body of the loop and must be indented — tipically 4 spaces — from the loop level. This indented level is a code block.
```

Comment:
```suggestion
`i` sequentially takes the value of the elements in `iterable`. It can have any valid name, but it's convenient to give it a meaningful name. `iterable` is the sequence you want to iterate over. Last but not least, the code to execute at each iteration constitutes the body of the loop and must be indented — typically 4 spaces — from the loop level. This indented level is a code block.

```